core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1129,6 +1138,8 @@ public void onRefreshSettings(Settings settings) {
             searchService.onRefreshSettings(settings);
             indexingService.onRefreshSettings(settings);
             if (change) {
+                engine().onSettingsChanged();
+                // TODO: why force a refresh here...?
```

Comment:
doesn't it follow the same logic as in updateBufferSize ? We want to have the buffer cleaned immediately if needed?